tools/testbench/topology.c

```diff
@@ -632,28 +652,34 @@ int load_process(void *dev, int comp_id, int pipeline_id,
 		printf("%u ", comp_ext.uuid[i]);
 	printf("\n");
 
-	/* Only one control is supported*/
-	if (widget->num_kcontrols > 1) {
-		fprintf(stderr, "error: more than one kcontrol defined\n");
-		return -EINVAL;
-	}
-
 	/* Get control into ctl and priv_data */
-	if (widget->num_kcontrols) {
+	for (i = 0; i < widget->num_kcontrols; i++) {
 		ret = tplg_load_one_control(&ctl, &priv_data, file);
 		if (ret < 0) {
 			fprintf(stderr, "error: failed control load\n");
 			return ret;
 		}
+
+		/* Merge process and priv_data into process_ipc */
+		if (priv_data)
+			ret = process_append_data(&process_ipc, &process, ctl, priv_data);
+		else
+			ret = 0;
```

Comment:
`ret` is already equal to 0 from `ret = tplg_load_one_control(&ctl, &priv_data, file);` above

Reply:
Done, thanks for improvements!